package/android/src/main/java/com/mrousavy/camera/CameraView+TakePhoto.kt

```diff
@@ -74,12 +73,19 @@ private fun writePhotoToFile(photo: CameraSession.CapturedPhoto, file: File) {
     val processedBitmap =
       Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, false)
     FileOutputStream(file).use { stream ->
-      processedBitmap.compress(Bitmap.CompressFormat.JPEG, 100, stream)
+      val didSucceed = processedBitmap.compress(Bitmap.CompressFormat.JPEG, 100, stream)
+      stream.close()
+      return if (didSucceed) imageBytes.size else 0
     }
   } else {
     val channel = FileOutputStream(file).channel
-    channel.write(byteBuffer)
+    val isReadyForWriting = byteBuffer.position() == 0
+    if (!isReadyForWriting) {
+      byteBuffer.flip()
```

Comment:
Did you test this codebranch? Does that work? I think if `position()` isn't `0` you need to call `rewind()` in the ByteBuffer. But I wouldn't even know when position isn't 0 tbh...